kong/clustering/wrpc_control_plane.lua

```diff
@@ -305,27 +307,35 @@ local function push_config_loop(premature, self, push_config_semaphore, delay)
     if exiting() then
       return
     end
-    if ok then
-      ok, err = pcall(self.push_config, self)
-      if ok then
-        local sleep_left = delay
-        while sleep_left > 0 do
-          if sleep_left <= 1 then
-            ngx.sleep(sleep_left)
-            break
-          end
-
-          ngx.sleep(1)
 
-          if exiting() then
-            return
+    if ok then
+      if next(self.clients) then
+        ok, err = pcall(self.push_config, self)
+        if ok then
+          local sleep_left = delay
+          while sleep_left > 0 do
+            if sleep_left <= 1 then
+              ngx.sleep(sleep_left)
+              break
+            end
+
+            ngx.sleep(1)
+
+            if exiting() then
+              return
+            end
+
+            sleep_left = sleep_left - 1
           end
 
-          sleep_left = sleep_left - 1
+        else
+          ngx_log(ngx_ERR, _log_prefix, "export and pushing config failed: ", err)
         end
 
       else
-        ngx_log(ngx_ERR, _log_prefix, "export and pushing config failed: ", err)
+        -- no clients connected, clear the stale config cache
+        self.config_call_rpc = nil
+        self.self.config_call_args = nil
```

Comment:
There's an extra `self` in here
```suggestion
        self.config_call_args = nil
```